tools/gyp/pylib/gyp/__init__.py

```diff
@@ -226,24 +235,24 @@ def Noop(value):
           (action == 'store_false' and not value)):
         flags.append(opt)
       elif options.use_environment and env_name:
-        print >>sys.stderr, ('Warning: environment regeneration unimplemented '
+        print('Warning: environment regeneration unimplemented '
                              'for %s flag %r env_name %r' % (action, opt,
-                                                             env_name))
+                                                             env_name), file=sys.stderr)
     else:
-      print >>sys.stderr, ('Warning: regeneration unimplemented for action %r '
-                           'flag %r' % (action, opt))
+      print('Warning: regeneration unimplemented for action %r '
+                           'flag %r' % (action, opt), file=sys.stderr)
 
   return flags
 
-class RegeneratableOptionParser(optparse.OptionParser):
-  def __init__(self):
+class RegeneratableOptionParser(argparse.ArgumentParser):
+  def __init__(self, usage):
     self.__regeneratable_options = {}
-    optparse.OptionParser.__init__(self)
+    argparse.ArgumentParser.__init__(self, usage=usage)
```

Comment:
Why is usage being passed through now when it wasn't before?

Reply:
Done in nodejs/node-gyp#1591 to print the __usage__ message if the user enters __-h__, etc.